lib/irmin.mli

```diff
@@ -2365,6 +2158,20 @@ module View (S: S): VIEW with type db = S.t
                           and type head = S.head
 (** Create views. *)
 
+val with_hrw_view :
+  (module S with type key = 'path) ->
+  (module VIEW with type t = 'view and type db = 'store and type key = 'path) ->
+  'store ->
```

Comment:
Instead, I've make `path` non-optional. Turns out that `Key.empty` was the only use of `module S` anyway, and it's quicker to pass an empty path than a store anyway.

Moving to `View` might make sense, but perhaps a job for another patch?